### PR TITLE
annotation set with image name and python dependency details

### PR DESCRIPTION
**Pull request by harshad16**
annotation set with image name and python dependency details
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
The UX design also shows a short description for each (lighter grey text):
image

@kywalker-rh Can you provide insight into where these come from or what they should be?

File: jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml [hunk 1]
```diff
@@ -5,7 +5,8 @@ metadata:
     opendatahub.io/notebook-image: "true"
   annotations:
     opendatahub.io/notebook-image-url: "https://github.com/thoth-station/s2i-generic-data-science-notebook"
-    opendatahub.io/notebook-image-name: "Generic Data Scient Notebook Image"
+    opendatahub.io/notebook-image-name: "Standard Data Science"
+    opendatahub.io/notebook-python-dependencies: [{"name":"Boto3","version":"1.17.11"},{"name":"Kafka-Python","version":"2.0.2"},{"name":"Matplotlib","version":"3.1.3"},{"name":"Numpy","version":"1.20.2"},{"name":"Pandas","version":"1.2.3"},{"name":"Scipy","version":"1.6.2"}]
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml:9]: Looking into this, I think this annotation needs to be on the `tags` level - the reason is that if we need to provide multiple versions of DS notebook, we need to make sure we can provide info for each of them - and they'd probably live in different tags - WDYT?

`spec.tags.annotations`:

* `opendatahub.io/notebook-python-dependencies`
* `opendatahub.io/notebook-tag-name`

I can imagine others like `cuda-compatibility` `verified-gpus` etc..

Comment (harshad16) [reply jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml:9]: i agree, we can have different versions.

Comment (kywalker-rh) [reply jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml:9]: @harshad16 Do you define somewhere what software and versions are in the notebook image as well as the python dependancies? I think that is what @jeff-philips-18 is looking for in this case.

Comment (harshad16) [reply jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml]: i have updated it now

**Comment by harshad16** [comment]
@kywalker-rh  @jeff-phillips-18  PTAL 
does this format works? list of dicts with name and version.

**Comment by jeff-phillips-18** [comment]
> @kywalker-rh @jeff-phillips-18 PTAL
> does this format works? list of dicts with name and version.

I'm not sure all entries are name/version pairs. One entry in the mocks shows `Provided by Anaconda`
@kywalker-rh thoughts?

**Comment by harshad16** [comment]
> I'm not sure all entries are name/version pairs. One entry in the mocks shows `Provided by Anaconda`

that would fit as well, [{"name":`Provided by Anaconda`, "version":""}]
in this way, on UI side you can directly use the list of object and set the value by e.name e.version

File: jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml [hunk 2]
```diff
@@ -5,14 +5,16 @@ metadata:
...
-    opendatahub.io/notebook-image-name: "Generic Data Scient Notebook Image"
+    opendatahub.io/notebook-image-name: "Standard Data Science"
     opendatahub.io/notebook-image-desc: "Jupyter notebook image with a set of data science libraries that advanced AI/ML notebooks will use as a base image to provide a standard for libraries avialable in all notebooks"
   name: s2i-generic-data-science-notebook
 spec:
   lookupPolicy:
     local: true
   tags:
   - annotations:
+      opendatahub.io/notebook-software: [{"name":"Python","version":"v3.8.3"}]
```
Comment (LaVLaS) [inline jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml:16]: I'm getting the error below when applying these annotations.  It looks like annotations values values can only be strings so these need to be single quoted in order to apply correctly

**Review by LaVLaS — CHANGES_REQUESTED** [review]

**Comment by jeff-phillips-18** [comment]
> > I'm not sure all entries are name/version pairs. One entry in the mocks shows `Provided by Anaconda`
> 
> that would fit as well, [{"name":`Provided by Anaconda`, "version":""}]
> in this way, on UI side you can directly use the list of object and set the value by e.name e.version

OK, that's fine then.

**Review by LaVLaS — APPROVED** [review]

**Comment by harshad16** [comment]
> OK, that's fine then.

i was just suggesting, please check on the UI side if this fits. we can modify this as per UI requirements

**Comment by jeff-phillips-18** [comment]
> > OK, that's fine then.
> 
> i was just suggesting, please check on the UI side if this fits. we can modify this as per UI requirements

I believe this will work for the UI.

**Pull request merged**